atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/transaction/impl/SnapshotTransaction.java

```diff
@@ -1862,12 +1881,44 @@ public void commit() {
     @Override
     public void commit(TransactionService transactionService) {
         try {
+            runPreCommitCallbacks();
             commitWithoutCallbacks(transactionService);
         } finally {
             close();
         }
     }
 
+    @com.palantir.lock.annotations.ReviewedRestrictedApiUsage
+    private void runPreCommitCallbacks() {
+        Map<TimestampLeaseName, PerLeaseActions> actions = preCommitActions.getActions();
```

Comment:
Very nice, especially the handling of the lock is now pretty clear. And the limiting long supplier, nice we could reu-use that.